lib/stellar_core_commander/commander.rb

```diff
@@ -37,12 +37,32 @@ def make_process
                           raise "Unknown process type: #{@process_type}"
                       end
 
-      process_class.new(tmpdir, base_port, identity, @process_options).tap do |p|
-        p.setup
+      process_class.new(transactor, tmpdir, name, base_port,
+                        identity, quorum, thresh, @process_options).tap do |p|
         @processes << p
       end
     end
 
+    Contract Transactor => Process
+    def get_root_process(transactor)
+      if @processes.size == 0
+        p = make_process transactor, :node0, [:node0], 1
+        @processes << p
+      end
+      @processes[0]
+    end
```

Comment:
More "let's show off ruby" comments below:

`if @processes.size == 0`

can also be written as:

`unless @processes.any?`

`any?` that returns true if any element of the collection matches the provided
predicate block:  `[1,2,4].any?{|n| n.odd? }` would return true, for example. If no block is provided, then a default block that returns true for any element is used, so you can IMO succinctly express "does the collection have any elements?" with `col.any?`

`@processes[0]` can also be expressed as `@processes.first`

---

These are all personal styling things, no need to change your work.  I just like showing off ruby some :)